homeassistant/components/template/alarm_control_panel.py

```diff
@@ -54,6 +56,16 @@
 CONF_DISARM_ACTION = "disarm"
 CONF_ALARM_CONTROL_PANELS = "panels"
 CONF_CODE_ARM_REQUIRED = "code_arm_required"
+CONF_CODE_FORMAT = "code_format"
+
+
+class CodeFormat(Enum):
+    """Class to represent different code formats."""
+
+    text = FORMAT_TEXT
+    number = FORMAT_NUMBER
+    none = None
+
```

Comment:
```suggestion

SUPPORTED_CODE_FORMATS = (FORMAT_TEXT, FORMAT_NUMBER, None)
```

Reply:
I tried this and it causes a validation error when wrong value is supplied. Are there any concerns about the current approach besides it being more verbose?
```
ERROR    homeassistant.config:config.py:906 Unknown error validating config for template platform for alarm_control_panel component with PLATFORM_SCHEMA
Traceback (most recent call last):
  File "/workspaces/home-assistant-core/homeassistant/config.py", line 895, in async_process_component_config
    p_validated = platform.PLATFORM_SCHEMA(p_config)  # type: ignore
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 386, in validate_mapping
    cval = cvalue(key_path, value)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 817, in validate_callable
    return schema(data)
  File "/workspaces/home-assistant-core/homeassistant/helpers/config_validation.py", line 489, in verify
    return cast(dict, schema(value))
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 386, in validate_mapping
    cval = cvalue(key_path, value)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 817, in validate_callable
    return schema(data)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 386, in validate_mapping
    cval = cvalue(key_path, value)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 817, in validate_callable
    return schema(data)
  File "/usr/local/lib/python3.9/site-packages/voluptuous/validators.py", line 741, in __call__
    'value must be one of {}'.format(sorted(self.container)))
TypeError: '<' not supported between instances of 'NoneType' and 'str'
```

Reply:
The ways of setting a yaml map item to what's `None` in Python is not very intuitive.
Let's do a string constant also for the no code case instead, perhaps by setting the `code_format` to `"no_code"` or `"codeless"`.

Reply:
Ok, will do. This will however require special casing that value while enum does it for us. Oh well...

Reply:
I just went back to enum implementation and all 4 of these are properly checked as invalid. That implementation didn't use nulls in yaml, it used `"none"` string, just like the current one.

Reply:
I still think "none" is potentially confusing.

While valid for YAML purposes, it will be confusing with things like JSON (which can be used in YAML) or confusing for people used to other programming languages.

I agree with the statement of renaming it to e.g., `no_code` or `codeless` instead of the current `none`.